terraform/a2ha-terraform/modules/aws_output/main.tf

```diff
@@ -27,4 +27,17 @@ resource "null_resource" "output" {
     provisioner "local-exec" {
         command = "mv ${path.root}/terraform.tfvars ${path.root}/aws.auto.tfvars;sed  -i 's/aws/deployment/' ${path.root}/.tf_arch;sed  -i 's/architecture \"aws\"/architecture \"deployment\"/' ${path.root}/../a2ha.rb"
     }
+
+    depends_on = [local_file.output]
 }
+
+#This resource will execute bash script to set up the destroy directory. Because we are maintaining single parent dir for terraform apply for provision and deployment. So just after provison, deploy command will be executed and for deployment diff terraform script will be executed. So we have to maintain destroy directory for aws resource deletion in future if needed.
+resource "null_resource" "bash" {
+    count = 1
+    
+    provisioner "local-exec" {
+        command = "cp /hab/a2_deploy_workspace/terraform/reference_architectures/aws/variables.tf /hab/a2_deploy_workspace/terraform/destroy/aws;cp /hab/a2_deploy_workspace/terraform/a2ha_habitat.auto.tfvars /hab/a2_deploy_workspace/terraform/destroy/aws/;cp /hab/a2_deploy_workspace/terraform/aws.auto.tfvars /hab/a2_deploy_workspace/terraform/destroy/aws;cp /hab/a2_deploy_workspace/terraform/variables_common.tf /hab/a2_deploy_workspace/terraform/destroy/aws"
```

Comment:
We could probably clean this up a bit. The destination is the same so we could probably make a list out of source files to copy and use the length for count. Would help with readability. Something like this:
```terraform
locals {
  copy_files = [
    "/hab/a2_deploy_workspace/terraform/reference_architectures/aws/variables.tf",
    "/hab/a2_deploy_workspace/terraform/a2ha_habitat.auto.tfvars",
    # all other paths
  ]
  destination_path = "/hab/a2_deploy_workspace/terraform/destroy/aws"
}

resource "null_resource" "setup_destroy_directory" {
  for_each = local.copy_files

  provisioner "local-exec" {
    command = "cp ${each.value} ${local.destination}"
  }
}
```

Either this or maybe just use `EOF` and separate out the `cp` command into multiple lines. Either of these would make it easier to read. Personally, I try to avoid `EOF` because you need to be mindful about indentation. Also, it doesn't look very pretty. Other than this, you can put those commands into a file and set the `command`'s value as return of the `file` call. 

However, the `locals` and `for_each` seem to do a pretty good job of keeping the relevant info at one place.

Reply:
Thanks Rishi for reviewing the code. I'll change accordingly.